misc/Kconfig

```diff
@@ -179,6 +179,13 @@ config BUILD_OUTPUT_BIN
 	  Build a binary in BIN format. This will build a zephyr.bin file need
 	  by some platforms.
 
+config BUILD_OUTPUT_EXE
+	bool "Build a binary in ELF format with .exe extension"
+	default n
+	help
+	  Build a binary in ELF format that can run in the host system. This
+	  will build a zephyr.exe file.
```

Comment:
i'm not sure a hard-coded .exe extension is such a good idea, .exe is strongly associated to most people with DOS/Windows, maybe call it "zephyr.native" or just "zephyr"?

Reply:
=> @nashif

Reply:
well, at least on windows the .exe would be accurate here, .exe is for executable, has really nothing to do with windows, in Linux this has been used in other environments (C# for example). Anyways, we can make this OS dependent, i.e. on windows it will be .exe, on macs it would be .app and on linux just zephyr. The idea with .exe was to make it immediately visible that this is a runnable binary on the host. (just calling it zephyr might not deliver this message)